### PR TITLE
CPM-117: Remove jstree on PEF and Mass Edit

### PR DESCRIPTION
**Pull request by pierallard**
This PR replaces the old "tree associate" module with a new one using Tree component from the DSM.
The "tree associate" renders trees in `<div id="tree-42">` elements, and stores the checked nodes in a hidden input.

This is not ideal, as
- we still use "id" to get children of the trees
- we still have an hidden input to store values
- we still have some jQuery for the tabs (Juju's working on it, I didn't want to touch this)
- we still have hardcoded ids in the DOM

BUT
This is better than before, and it does not use the jstree, and I didn't tire la pelote de laine

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+const Router = require('pim/router');
+
+type CategoryResponse = {
+  attr: {
+    id: string;
+    'data-code': string;
+  };
+  children?: CategoryResponse[];
+  data: string;
+  state: string; // 'closed jstree-root' | 'leaf' | 'closed' | 'leaf toselect jstree-checked';
```
Comment (mmetayer) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx:16]: Not worth creating a type for that?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+            childrenCallback={childrenCallback}
+            init={init}
+          />
+        </ThemeProvider>
+      </DependenciesProvider>,
+      tree
+    );
+  }
+
+  private getChildrenRoute = (id: number) => {
```
Comment (mmetayer) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx:181]: this is more an URL (generated from a route)

File: tests/legacy/features/Context/Page/Product/Edit.php [hunk 3]
```diff
@@ -232,7 +232,7 @@ protected function extractLabelElement($field, ElementInterface $element = null)
 
         if (strstr($field, 'USD') || strstr($field, 'EUR')) {
             if (false !== strpos($field, ' ')) {
-                list($subLabelContent, $labelContent) = explode(' ', $field);
+                [$subLabelContent, $labelContent] = explode(' ', $field);
```
Comment (mmetayer) [inline tests/legacy/features/Context/Page/Product/Edit.php:235]: :heart:

**Review by mmetayer — APPROVED** [review]
Is it possible to add unit tests for the new components?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTree.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { RecursiveCategoryTree } from "./RecursiveCategoryTree";
+import {Tree} from 'akeneo-design-system/lib/components/Tree/Tree';
```
Comment (nmarniesse) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTree.tsx:3]: `import {A}` and `import { A }` are both validated by prettier?

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/RecursiveCategoryTree.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    label={treeState.label}
+    value={{
+      id: treeState.id,
+      code: treeState.code
+    }}
+    selected={treeState.selected}
+    isLoading={treeState.loading}
+    readOnly={treeState.readOnly}
+    selectable={treeState.selectable}
+    isLeaf={Array.isArray(treeState.children) && treeState.children.length == 0}
```
Comment (nmarniesse) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/RecursiveCategoryTree.tsx:47]:
```suggestion
    isLeaf={Array.isArray(treeState.children) && treeState.children.length === 0}
```

File: tests/legacy/features/Behat/Decorator/Tree/TreeDecorator.php [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+    public function select()
+    {
+        $checkbox = $this->find('css', 'div[role=checkbox]');
+        if ($checkbox->getAttribute('aria-checked') === 'false') {
+            $checkbox->click();
+        }
+    }
+
+    public function expandNode($nodeName)
```
Comment (nmarniesse) [inline tests/legacy/features/Behat/Decorator/Tree/TreeDecorator.php:41]:
```suggestion
    public function expandNode(string $nodeName): void
```

**Review by nmarniesse — APPROVED** [review]
Nice job :+1:

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx (hunk 1 above)
Comment (pierallard) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/TreeAssociate.tsx:16]: I don't know how to put a type, it's a composition of strings separated by spaces :/

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTree.tsx (hunk 4 above)
Comment (pierallard) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/workspaces/shared/src/components/CategoryTree/CategoryTree.tsx:3]: I forgot to run prettier :)

**Pull request merged**